src/wp-includes/fonts.php

```diff
@@ -107,31 +125,77 @@ function wp_unregister_font_collection( string $slug ) {
  *     @type string|false $error   False or error message.
  * }
  */
-function wp_get_font_dir() {
+function wp_font_dir( $create_dir = true ) {
+	/*
+	 * Allow extenders to manipulate the font directory consistently.
+	 *
+	 * Ensures the upload_dir filter is fired both when calling this function
+	 * directly and when the upload directory is filtered in the Font Face
+	 * REST API endpoint.
+	 *
+	 * The filter runs at priority 20 to ensure it runs after any filters
+	 * added at the default priority by plugins modifying the upload path
+	 * or URL.
+	 */
+	add_filter( 'upload_dir', 'wp_apply_font_dir_filter', 20 );
+	$font_dir = wp_upload_dir( null, $create_dir, false );
+	remove_filter( 'upload_dir', 'wp_apply_font_dir_filter', 20 );
+	return $font_dir;
+}
+
+/**
+ * Applies filters to manipulate the font directory used by the font library.
+ *
+ * Wrapper for apply_filters( 'font_dir', $font_dir ).
+ * Does not apply filters if already doing the {@see 'font_dir'} filter.
+ *
+ * Allows developers to manipulate the font directory used by the
+ * font library.
+ *
+ * @since 6.5.0
+ *
+ * @param string $font_dir The font directory.
+ * @return string The modified font directory.
+ */
+function wp_apply_font_dir_filter( $font_dir ) {
```

Comment:
I've been thinking about the name of this function. While still not sure on the general name, I'd suggest that we at least call it `wp_apply_font_dir_filters()` to reflect the docblock, and the fact it applies _all_ `'font_dir'`filters.

Just brainstorming here, but what about `wp_get_filtered_font_dir()`?

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/6211/commits/52103f63b6905f3bcb980303b3bfa9905a58c9d2